ui/guest/src/util.tsx

```diff
@@ -83,25 +86,25 @@ export const EditingStatus = {
   SHOW_RECEPTACLES: 'SHOW_RECEPTACLES'
 };
 
-export function notNullOrUndefined(value) {
+export function notNullOrUndefined(value: any): boolean {
   return value != null;
 }
 
-export function isNullOrUndefined(value) {
+export function isNullOrUndefined(value: any): boolean {
   return value == null;
 }
 
-export function not(condition) {
+export function not(condition: boolean): boolean {
   return !condition;
 }
 
-export function sibling(element, next) {
+export function sibling(element: HTMLElement, next: boolean) {
   return (next)
     ? element.nextElementSibling
     : element.previousElementSibling;
 }
 
-export function forEach(array, fn, emptyReturnValue) {
+export function forEach(array: any[], fn: Function, emptyReturnValue?) {
```

Comment:
- Return type
- Type function properly to express possibilities ('break', 'continue', specific value, etc)
- Type emptyReturnValue